cmake/mkl/MKLConfig.cmake

```diff
@@ -208,12 +208,28 @@ if(CMAKE_Fortran_COMPILER)
 endif()
 
 # Determine Compiler Family
-if(CXX_COMPILER_NAME STREQUAL "dpcpp" OR CXX_COMPILER_NAME STREQUAL "dpcpp.exe"
-    OR CXX_COMPILER_NAME STREQUAL "icpx" OR CXX_COMPILER_NAME STREQUAL "icx.exe")
-  set(SYCL_COMPILER ON)
+
+include(CMakePackageConfigHelpers)
+include(CheckCXXCompilerFlag)
+include(CheckIncludeFileCXX)
+include(GNUInstallDirs)
+
+# Check SYCL support by the compiler
+check_cxx_compiler_flag("-fsycl" _fsycl_option)
+if (_fsycl_option)
+  CHECK_INCLUDE_FILE_CXX("sycl/sycl.hpp" _sycl_header "-fsycl")
+  if (NOT _sycl_header)
+    CHECK_INCLUDE_FILE_CXX("CL/sycl.hpp" _sycl_header_old "-fsycl")
```

Comment:
`SYCL_SYCL_HPP` and `CL_SYCL_HPP` might be more expressive than `_sycl_header` and `_sycl_header_old`. It would be great if you also use `COMPILER_FSYCL_OPTION` instead of `_fsycl_option`.